modules/project/alerts.tf

```diff
@@ -0,0 +1,165 @@
+          alignment_period     = try(each.value.conditions.condition_absent.aggregations.alignment_period, null)
+        }
+        duration = each.value.conditions.condition_absent.duration
+      }
+    }
+    dynamic "condition_threshold" {
+      for_each = lookup(each.value.conditions, "condition_threshold", null)[*]
+      content {
+        threshold_value    = try(each.value.condition_threshold.threshold_value, null)
+        denominator_filter = try(each.value.condition_threshold.denominator_filter, null)
+        dynamic "denominator_aggregations" {
+          for_each = try(each.value.condition_threshold.denominator_aggregations, null)[*]
+          content {
+            per_series_aligner = try(each.value.conditions.condition_threshold.denominator_aggregations.per_series_aligner, null)
+            group_by_fields    = try(each.value.conditions.condition_threshold.denominator_aggregations.group_by_fields, [])
+            alignment_period   = try(each.value.conditions.condition_threshold.denominator_aggregations.alignment_period, null)
+          }
+        }
+        dynamic "forecast_options" {
+          for_each = try(each.value.conditions.condition_threshold.forecast_options, null)[*]
+          content {
+            forecast_horizon = try(each.value.conditions.condition_threshold.forecast_options.forecast_horizon, null)
+          }
+        }
+        comparison              = try(each.value.conditions.condition_threshold.comparison, null)
+        duration                = try(each.value.conditions.condition_threshold.duration, null)
+        evaluation_missing_data = try(each.value.conditions.condition_threshold.evaluation_missing_data, null)
+        filter                  = try("resource.type = \"${each.value.conditions.condition_threshold.resource_type}\" AND metric.type = \"logging.googleapis.com/user/${google_logging_metric.default[each.value.name].name}\"", each.value.conditions.condition_threshold.filter)
+        trigger {
+          count   = try(each.value.conditions.condition_threshold.trigger.count, null)
+          percent = try(each.value.conditions.condition_threshold.trigger.percent, null)
+        }
+        aggregations {
+          per_series_aligner   = try(each.value.conditions.condition_threshold.aggregations.per_series_aligner, null)
+          group_by_fields      = try(each.value.conditions.condition_threshold.aggregations.group_by_fields, [])
+          cross_series_reducer = try(each.value.conditions.condition_threshold.aggregations.cross_series_reducer, null)
+          alignment_period     = try(each.value.conditions.condition_threshold.aggregations.alignment_period, null)
+        }
+      }
+    }
+  }
+  notification_channels = [
+    try(google_monitoring_notification_channel.this[each.value.notification_channels].id,
+    each.value.notification_channels),
+    try(google_monitoring_notification_channel.default[0].id, "")
+  ]
```

Comment:
I'm not sure this works (you're indexing `google_monitoring_notification_channel.this` with `each.value.notification_channels`, which is a list). Please double check.

Also, I think the default shouldn't be used if there are notification channels defined by the alert. For me, the logic should be: 
1. Try to use the locally created channel first,
2. If the specified channel id is not created by us, use it as as an id (i.e. the notification channel was created elsewhere)
3. If there are no notification channels, use the default, if provided.